src/default_fetcher.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 David Waite
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import 'whatwg-fetch';
+import nodeFetcher from 'node-fetch';
+
+let fetcher: GlobalFetch;
```

Comment:
I don't want to rely on a global `fetch` for the dependency resolution. I was specifically trying to avoid this.

I want the `Requestor` interface to exist, because it is easy to provide implementations of the interface where the types expect the dependency to be present (rather than a global which has implicit races on whether such a global was exported correctly).

Also, I *only* want to change the underlying typings in the requestor to rely on the lower level types in `whatwg-fetch` (because they are possibly easier to reason about and developers are probably going to be more comfortable with them).

Investigating this further, it looks like the typings for `whatwg-fetch` (as they are currently written) are not usable without depending on a global `fetch`. i.e. the underlying lower level types like `HeadersInit` and `RequestInfo` etc. are not `export`-ed at all. We should send them a PR to fix that first.

Sorry about sending you in the wrong direction. I hope you understand what i meant now 😄.